Source/SA/RFmx SpecAn/RFmxSpecAn.cs

```diff
@@ -44,33 +40,313 @@ public struct AutoLevelConfiguration
+            public double MinimumPower_dBm;
+        }
+        public struct AcpOffsetResults
+        {
+            public double LowerAbsolutePower_dBm;
+            public double LowerRelativePower_dB;
+            public double UpperAbsolutePower_dBm;
+            public double UpperRelativePower_dB;
+            public double OffsetFrequency_Hz;
+            public double OffsetIntegrationBandwidth_Hz;
+        }
+
+        public struct AcpComponentCarrierResults
+        {
+            public double CarrierFrequency_Hz;
+            public double IntegrationBandwidth_Hz;
+            public double AbsolutePower_dBm_or_dBmHz;
+            public double TotalRelativePower_dB;
+        }
+
+        public struct AcpResults
+        {
+            public AcpOffsetResults[] OffsetResults;
+            public AcpComponentCarrierResults[] ComponentCarrierResults;
+        }
+        public struct AmpmResults
+        {
+            public double MeanLinearGain_dB;
+            public double OnedBCompressionPoint_dBm;
+            public double MeanRmsEvm_percent;
+            public double AmToAMResidual;
+            public double AmToPMResidual;
+        }
+
+
         #endregion
         #region Instrument Configurations
         public static void ConfigureCommon(RFmxInstrMX sessionHandle, RFmxSpecAnMX specAnSignal, CommonConfiguration commonConfig,
             AutoLevelConfiguration autoLevelConfig, string selectorString = "")
         {
             sessionHandle.ConfigureFrequencyReference("", commonConfig.FrequencyReferenceSource, 10e6);
-            sessionHandle.SetLOSource("", commonConfig.LOSource);
-            sessionHandle.SetDownconverterFrequencyOffset("", commonConfig.LOOffset);
             specAnSignal.SetSelectedPorts(selectorString, commonConfig.SelectedPorts);
             specAnSignal.ConfigureDigitalEdgeTrigger(selectorString, commonConfig.DigitalEdgeSource, commonConfig.DigitalEdgeType, commonConfig.TriggerDelay_s, commonConfig.EnableTrigger);
             specAnSignal.ConfigureFrequency(selectorString, commonConfig.CenterFrequency_Hz);
-            specAnSignal.Spectrum.Configuration.ConfigureSpan(selectorString, commonConfig.Span_Hz);
             specAnSignal.ConfigureExternalAttenuation(selectorString, commonConfig.ExternalAttenuation_dB);
-
-            if (autoLevelConfig.AutoLevelReferenceLevel) specAnSignal.AutoLevel(selectorString, commonConfig.Span_Hz, autoLevelConfig.AutoLevelMeasureTime_s, out _);
+            if (autoLevelConfig.AutoLevelReferenceLevel) specAnSignal.AutoLevel(selectorString, autoLevelConfig.AutoLevelBandwidth_Hz, autoLevelConfig.AutoLevelMeasureTime_s, out _);
             else specAnSignal.ConfigureReferenceLevel(selectorString, commonConfig.ReferenceLevel_dBm);
 
         }
+        public static void ConfigureTxp(RFmxSpecAnMX specAn, TxpConfiguration txpConfig, string selectorString = "")
+        {
+            specAn.Txp.Configuration.SetMeasurementEnabled(selectorString, true);

```

Comment:
enable traces too

Reply:
will to for alignment reason but we should disable traces by default as we talk about automation